### PR TITLE
HBASE-27998 Enhance region metrics to include prefetch ratio for each…

### PR DESCRIPTION
**Pull request by ragarkar**
… region

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 45s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   3m 34s |  HBASE-27389 passed  |
| +1 :green_heart: |  checkstyle  |   0m 52s |  HBASE-27389 passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m  8s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 30s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 13s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool |
| uname | Linux aef0902bf807 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 3a2333e6b5 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 81 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 26s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 17s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 12s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 51s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 24s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 21s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 224m 18s |  hbase-server in the patch failed.  |
|  |   | 249m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 88a54a1b4b57 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 3a2333e6b5 |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/testReport/ |
| Max. process+thread count | 4878 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 15s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 50s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   5m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 53s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 35s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 39s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 35s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 228m 58s |  hbase-server in the patch failed.  |
|  |   | 258m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux a08ccf42e7d3 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 3a2333e6b5 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/testReport/ |
| Max. process+thread count | 4333 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ServerMetrics.java [hunk 1]
```diff
@@ -106,4 +106,6 @@ default String getVersion() {
   @Nullable
   List<ServerTask> getTasks();
 
+  /** Returns the region prefetch information for the regions hosted on this server */
+  Map<String, Integer> getRegionPrefetchInfo();
```
Comment (wchevreuil) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/ServerMetrics.java:110]: Nit: Mind explain in the comment what is the Integer value representing here? Is it the total region size, or amount of the region size cached? If the former, is that absolute or round percentage?

**Review by wchevreuil — COMMENTED** [review]
LGTM, just a minor nit regarding comments.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/ServerMetrics.java (hunk 1 above)
Comment (ragarkar) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/ServerMetrics.java]: Updated the comment to add information about about the integer value in the regionPrefetchInfo.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 50s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   5m  6s |  HBASE-27389 passed  |
| +1 :green_heart: |  checkstyle  |   1m 14s |  HBASE-27389 passed  |
| -1 :x: |  spotless  |   0m 55s |  branch has 65 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   6m 35s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   5m 32s |  the patch passed  |
| +1 :green_heart: |  cc  |   5m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   5m 32s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  8s |  The patch passed checkstyle in hbase-protocol-shaded  |
| +1 :green_heart: |  checkstyle  |   0m 17s |  The patch passed checkstyle in hbase-client  |
| +1 :green_heart: |  checkstyle  |   0m 41s |  hbase-server: The patch generated 0 new + 17 unchanged - 3 fixed = 17 total (was 20)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m  8s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 20s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 29s |  The patch does not generate ASF License warnings.  |
|  |   |  55m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 751c70b030fa 5.4.0-156-generic #173-Ubuntu SMP Tue Jul 11 07:25:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-general-check/output/branch-spotless.txt |
| Max. process+thread count | 80 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by wchevreuil — APPROVED** [review]
LGTM, just need to address the spotless errors before this can be merged.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 13s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 17s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   4m 55s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 26s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 21s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 231m 17s |  hbase-server in the patch failed.  |
|  |   | 256m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 351210cfe9e7 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/testReport/ |
| Max. process+thread count | 4780 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 47s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 46s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   4m 39s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 46s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 46s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 40s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 40s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 34s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 232m  1s |  hbase-server in the patch failed.  |
|  |   | 260m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 132a8bb9dd26 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/testReport/ |
| Max. process+thread count | 4537 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by wchevreuil** [comment]
re-check

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  prototool  |   0m  0s |  prototool was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 15s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 48s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   3m 36s |  HBASE-27389 passed  |
| +1 :green_heart: |  checkstyle  |   0m 50s |  HBASE-27389 passed  |
| -1 :x: |  spotless  |   0m 37s |  branch has 65 errors when running spotless:check, run spotless:apply to fix.  |
| +1 :green_heart: |  spotbugs  |   4m 11s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 33s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 31s |  the patch passed  |
| +1 :green_heart: |  cc  |   3m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 31s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m  5s |  The patch passed checkstyle in hbase-protocol-shaded  |
| +1 :green_heart: |  checkstyle  |   0m 13s |  The patch passed checkstyle in hbase-client  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  hbase-server: The patch generated 0 new + 17 unchanged - 3 fixed = 17 total (was 20)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |   9m 40s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.5.  |
| +1 :green_heart: |  hbaseprotoc  |   1m 17s |  the patch passed  |
| +1 :green_heart: |  spotless  |   0m 39s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   4m 40s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile cc hbaseprotoc prototool |
| uname | Linux 3f5768c0e2f1 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| spotless | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/artifact/yetus-general-check/output/branch-spotless.txt |
| Max. process+thread count | 80 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 12s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 18s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   4m 53s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 17s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 25s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 20s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 228m 10s |  hbase-server in the patch passed.  |
|  |   | 253m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 449ce452d48f 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/testReport/ |
| Max. process+thread count | 4536 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27389 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 53s |  HBASE-27389 passed  |
| +1 :green_heart: |  compile  |   1m 52s |  HBASE-27389 passed  |
| +1 :green_heart: |  shadedjars  |   4m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  HBASE-27389 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 52s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 38s |  hbase-protocol-shaded in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 36s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 235m 29s |  hbase-server in the patch failed.  |
|  |   | 264m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5342 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 1da89b2f8497 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27389 / 2469aca027 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/testReport/ |
| Max. process+thread count | 4585 (vs. ulimit of 30000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5342/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by ragarkar** [comment]
> LGTM, just need to address the spotless errors before this can be merged.

Not sure why there are spotless error here. I do not see any errors in my local build. The 'mvn spotless:check' comes out clean.

**Comment by ragarkar** [comment]
re-check

**Comment by wchevreuil** [comment]
The latest UT failures seem unrelated. I can't reproduce the spotless failures, so let's get this merged.

**Pull request merged**